### PR TITLE
dev-util/amdapp: Added release and manifest for new AMD download v3.0.130.136

### PR DESCRIPTION
**Pull request by cengique**
Package-Manager: portage-2.2.28

### CODE REVIEW

File: dev-util/amdapp/amdapp-3.0-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
```
Comment (jlec) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Please use EAPI=6

File: dev-util/amdapp/amdapp-3.0-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2015 Gentoo Foundation
```
Comment (jlec) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: 2016

**Comment by jlec** [comment]
You should use ```repoman```.

File: dev-util/amdapp/amdapp-3.0-r2.ebuild (hunk 1 above)
Comment (cengique) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Done

File: dev-util/amdapp/amdapp-3.0-r2.ebuild (hunk 2 above)
Comment (cengique) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Done

**Comment by cengique** [comment]
I was using `repoman` - maybe I don't know how to interpret its output. Did you mean that the EAPI deprecation was flagged there? Here's what I get from `repoman -vd full`:
```sh
$ repoman -vd full

RepoMan scours the neighborhood...
[INFO] checking package dev-util/amdapp
  repo.eapi.deprecated          2
   dev-util/amdapp/amdapp-2.9.1-r1.ebuild: 5
   dev-util/amdapp/amdapp-3.0-r1.ebuild: 5
RepoMan sez: "You're only giving me a partial QA payment?
              I'll take it this time, but I'm not happy."```

My commit message is also not the best, but I'll try to do better next time. :/

Why is Travis failing? Is there anything to do with my changes? It's complaining about `.travis.yml` being out of date among other things that I don't understand.

File: dev-util/amdapp/amdapp-3.0-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+src_compile() {
+	MAKEOPTS+=" -j1"
+	use examples && cd samples/opencl && default
+}
+
+src_install() {
+	dodir /opt/AMDAPP
+	cp -R "${S}/"* "${ED}/opt/AMDAPP" || die "Install failed!"
```
Comment (jlec) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: why not ```doins -r```?

Comment (cengique) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Not sure - I just copied from the previous ebuild.

Comment (marbre) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: @cengique Please replace the `cp -R` as suggested. Sometimes things aren't solved the way they should be in _older_ ebuilds.

**Comment by cengique** [comment]
@marbre thanks for explaining what the previous poster meant. I made the change and tested the ebuild locally. The commit message was truncated because I had backticks in it. I guess it's too late to amend it because it's already pushed here?

**Comment by cengique** [comment]
@marbre I rebased onto master to be up-to-date. If this is too messy, I can start over and create a new pull request.

**Comment by marbre** [comment]
@cengique Actually it seems something went wrong with rebasing and yes it is a little messy. Anyway, a new pull request isn't necessary. One possibility is to save your changes (e.g. using git format-patch), reset --hard to the sci/master branch, apply your changes and push --force to your existing `dev-util/amdapp` branch. Another is to delete the `dev-util/amdapp` branch, create a new `dev-util/amdapp` branch, apply your changes and push force afterwards.

Further, please use `repoman commit` for every commit and don't delete the package name from the commit message, as correctly formated commit messages (e.g. _dev-util/amdapp: this commit does this and that_) are mandatory.

**Comment by cengique** [comment]
@marbre Thanks for the explanation. I did as you suggested, and squashed everything into one commit. But I guess because a lot of time has passed, repoman on TravisCI is giving errors that I'm not getting locally. I will try to upgrade my portage. 

In the meantime, do you know what the `dependency.bad [fatal]        52` error from repoman means?

**Comment by marbre** [comment]
@cengique `x11-drivers/ati-drivers` is masked, thus the dependency needs to be replaced by the appropriate package now providing the drivers.

**Comment by marbre** [comment]
@cengique Check commit 14a2831.

**Comment by cengique** [comment]
@marbre @jlec Thanks, I removed the dependency for `x11-drivers/ati-drivers` and checks seem to pass now.

**Comment by cengique** [comment]
Note: I did use `repoman commit -m "..."`, not sure why the `dev-utils/amdapp` prefix is missing in the commit message.

**Comment by cengique** [comment]
Oops, apparently I messed up `src_install` when I used `doins`. See my last commit. @jlec `doins` is only for single files and does not replace `cp -r` in this case. Btw, the ebuild help documentation is really difficult to understand with so few examples.

**Comment by jlec** [comment]
@cengique `doins -r` is what you want. Check the [devmanual](https://devmanual.gentoo.org) for more information and examples.

**Comment by cengique** [comment]
@jlec Ok, I see now that `doins -r` exists. It was in the previous version because apparently you suggested it to me, but after checking the devmanual, I was unable to find any references to it and deleted it. Neither [dev-manual/src_install](https://devmanual.gentoo.org/ebuild-writing/functions/src_install/index.html) or [devmanual/install-functions](https://devmanual.gentoo.org/function-reference/install-functions/index.html) has any reference to `doins -r`, so it should be added somewhere or it's very confusing to n00bs like me!

So now I'm back to using `doins -r` and also added a few more commands to set executable bits and remove unpacked archive.

**Comment by cengique** [comment]
@jlec @marbre is everything good with the patch now? We're about to elapse a full year since this request first opened. I hope we can get it closed soon.

File: dev-util/amdapp/amdapp-3.0-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,91 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
```
Comment (jlec) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Please drop this line

File: dev-util/amdapp/amdapp-3.0-r2.ebuild [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+src_compile() {
+	MAKEOPTS+=" -j1"
+	use examples && cd samples/opencl && default
+}
+
+src_install() {
+	# Copy everything
+	dodir $OPT_DIR
+	insinto $OPT_DIR
+	doins -r * || die "Install failed!"
```
Comment (jlec) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Drop `die`

File: dev-util/amdapp/amdapp-3.0-r2.ebuild [hunk 6]
```diff
@@ -68,6 +70,22 @@ src_compile() {
 }
 
 src_install() {
-	dodir /opt/AMDAPP
-	cp -r "${S}/"* "${ED}/opt/AMDAPP" || die "Install failed!"
+	# Copy everything
+	dodir $OPT_DIR
```
Comment (cengique) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Done

File: dev-util/amdapp/amdapp-3.0-r2.ebuild [hunk 7]
```diff
@@ -68,6 +70,22 @@ src_compile() {
...
+	insinto $OPT_DIR
+	doins -r * || die "Install failed!"
```
Comment (cengique) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Done

File: dev-util/amdapp/amdapp-3.0-r2.ebuild [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+AMD64_AT="AMD-APP-SDKInstaller-v${MY_V}-linux64.tar.bz2"
+
+MY_P_AMD64="AMD-APP-SDK-v${MY_V}-linux64.sh"
+MY_P_AMD32="AMD-APP-SDK-v${MY_V}-linux32.sh"
+
+DESCRIPTION="AMD Accelerated Parallel Processing (APP) SDK"
+HOMEPAGE="http://developer.amd.com/tools-and-sdks/opencl-zone/amd-accelerated-parallel-processing-app-sdk"
+SRC_URI="
+		amd64? ( ${AMD64_AT} )
+		x86? ( ${X86_AT} )"
```
Comment (cengique) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Done

File: dev-util/amdapp/amdapp-3.0-r2.ebuild [hunk 9]
```diff
@@ -68,6 +70,22 @@ src_compile() {
...
+	# Set executable bits
+	exeinto $OPT_DIR/bin/x86_64/
+	doexe bin/x86_64/clinfo
+
+	exeinto $OPT_DIR/bin/x86/
+	doexe bin/x86/clinfo
+
+	# Delete archive - already unpacked
+	if use amd64 || use amd64-linux ; then
+		rm "${D}/${OPT_DIR}/${MY_P_AMD64}"
```
Comment (cengique) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Done; although with no message.

File: dev-util/amdapp/amdapp-3.0-r2.ebuild [hunk 10]
```diff
@@ -68,6 +70,22 @@ src_compile() {
...
+	else
+		rm "${D}/${OPT_DIR}/${MY_P_X86}"
```
Comment (cengique) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Done

File: dev-util/amdapp/amdapp-3.0-r2.ebuild [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+src_compile() {
+	MAKEOPTS+=" -j1"
+	use examples && cd samples/opencl && default
+}
+
+src_install() {
+	dodir /opt/AMDAPP
+	doins -r "${S}/"* "${ED}/opt/AMDAPP" || die "Install failed!"
```
Comment (cengique) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: This seems to be from an older version, it's already fixed.

File: dev-util/amdapp/amdapp-3.0-r2.ebuild (hunk 4 above)
Comment (cengique) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Done

File: dev-util/amdapp/amdapp-3.0-r2.ebuild (hunk 5 above)
Comment (cengique) [inline dev-util/amdapp/amdapp-3.0-r2.ebuild]: Done

**Comment by cengique** [comment]
@jlec I think I fixed all the issues you raised, however Travis fails because I seem to have an older copy of `.travis.yml` AFAIK. Last time I rebased it caused a lot of conflicts, so please check before I have to do that.

**Pull request merged**